src/core_modules/capture-core/components/Pages/Search/SearchForm/SearchForm.epics.js

```diff
@@ -160,3 +162,94 @@ export const searchViaAttributesOnScopeTrackedEntityTypeEpic: Epic = (action$, s
             return searchViaAttributesStream(queryArgs, attributes, triggeredFrom);
         }),
     );
+
+const deriveSearchFormInfo = searchGroups => (searchGroups.filter(searchGroup => !searchGroup.unique)[0] || {});
+
+// falling back from a search into a Program to a search into TEType means that
+// sometimes there wil be less attributes to search with. For instance a program
+// can have attributes last name, first name and gender but a TETYpe will have
+// only first name. Here we derive the form values that are revelant.
+const deriveFormValues = (searchForm, values) => {
+    if (!searchForm) {
+        return {};
+    }
+    const elements = searchForm.getElements();
+
+    return ([...elements.values()].reduce((acc, { id: fieldId }) => {
+        if (Object.keys(values).includes(fieldId)) {
+            const fieldValue = values[fieldId];
+            return { ...acc, [fieldId]: fieldValue };
+        }
+        return acc;
+    }, {}));
+};
+
+const deriveCurrentFallbackSearchTerms = (searchTermsFromOriginalSearch, fallbackFormValues) =>
+    searchTermsFromOriginalSearch.filter(({ id }) => Object.keys(fallbackFormValues).includes(id));
+
+export const startFallbackSearchEpic = (action$: InputObservable, store: ReduxStore) =>
+    action$.pipe(
+        ofType(searchPageActionTypes.FALLBACK_SEARCH_START),
+        flatMap(({ payload: { formId, programId, pageSize, availableSearchOptions } }) => {
+            const trackerProgram = getTrackerProgramThrowIfNotFound(programId);
+            if (trackerProgram.trackedEntityType) {
+                const { orgUnitId } = store.value.currentSelections;
+
+                const { id: trackedEntityTypeId } = trackerProgram.trackedEntityType;
+                const { formsValues, searchPage: { currentSearchInfo: { currentSearchTerms: searchTermsFromOriginalSearch } } } = store.value;
+
+                const { searchForm, formId: fallbackFormId } = deriveSearchFormInfo(availableSearchOptions[trackedEntityTypeId].searchGroups);
+                const fallbackFormValues = deriveFormValues(searchForm, formsValues[formId]);
+                const fallbackSearchTerms = deriveCurrentFallbackSearchTerms(searchTermsFromOriginalSearch, fallbackFormValues);
+
+                return concat(
+                    of(fallbackPushPage({ orgUnitId, trackedEntityTypeId })),
+                    of(fallbackSearch({ trackedEntityTypeId, fallbackFormValues, pageSize })),
+                    of(saveCurrentSearchInfo({
```

Comment:
Here we are having three actions being dispatched. 
One that pushes to the new page. 
One that saves the current search information such us search terms. We do this because we want the form in the new page to be prepopulated with the same values the user has already typed.
And last we trigger the fallbackSearch which does the actual fetching of the users.